File: cpp/space-age/space_age.cpp
#include "space_age.h"

namespace space_age {
    space_age::space_age(long int seconds) {
        _seconds = seconds;
        noOfEarthYears = double(seconds)/31557600;
    }
    long int space_age::seconds() const {
        return _seconds;
    }
    double space_age::on_mercury() const {
        return noOfEarthYears/planets.at("Mercury");
    }
    double space_age::on_venus() const {
        return noOfEarthYears/planets.at("Venus");
    }
    double space_age::on_earth() const {
        return noOfEarthYears;
    }
    double space_age::on_mars() const {
        return noOfEarthYears/planets.at("Mars");
    }
    double space_age::on_jupiter() const {
        return noOfEarthYears/planets.at("Jupiter");
    }
    double space_age::on_saturn() const {
        return noOfEarthYears/planets.at("Saturn");
    }
    double space_age::on_uranus() const {
        return noOfEarthYears/planets.at("Uranus");
    }
    double space_age::on_neptune() const {
        return noOfEarthYears/planets.at("Neptune");
    }
}  // namespace space_age

File: go/space-age/space_age.go
package space

type Planet string

var multiplier = map[Planet]float64 {
    "Mercury": 0.2408467,
    "Venus": 0.61519726,
    "Earth": 1,
    "Mars": 1.8808158,
    "Jupiter": 11.862615,
    "Saturn": 29.447498,
    "Uranus": 84.016846,
    "Neptune": 164.79132,
}

// Returns age of person on a particular planet,
// given age in seconds and planet name
func Age(ageInSecond float64, planet Planet) float64 {
    earthAge := ageInSecond/float64(60*60*24*365.2425)
    return earthAge/multiplier[planet]
}
